src/hooks/use-intersection-observer.ts

```diff
@@ -1,9 +1,9 @@
 import * as React from 'react'
 
-export const useIntersectionObserver = (
+export const useIntersectionObserver = <T extends HTMLElement>(
```

Comment:
Nice! `Element` va mejor porque también soporta `SVG` elements!

```suggestion
export const useIntersectionObserver = <T extends Element>(
```